elephant/spike_train_correlation.py

```diff
@@ -951,13 +954,16 @@ def spike_train_timescale(binned_st, tau_max):
     """
     binsize = binned_st.binsize
     if not (tau_max / binsize).simplified.units == pq.dimensionless:
-        raise AssertionError("tau_max needs units of time")
+        raise ValueError("tau_max needs units of time")
 
     # safe casting of tau_max/binsize to integer
     tau_max_bins = int(np.round((tau_max / binsize).simplified.magnitude))
     if not np.isclose(tau_max.simplified.magnitude,
                       (tau_max_bins * binsize).simplified.magnitude):
-        raise AssertionError("tau_max has to be a multiple of the binsize")
+        raise ValueError("tau_max has to be a multiple of the binsize")
+
+    if len(np.where(binned_st.to_array())[1]) <= 2 and with_nan:
```

Comment:
a more efficient and clear version is
`if with_nan and binned_st.get_num_of_spikes() <= 2`

However, I'm not sure that we should return NaN _only_ when `with_nan` is specified. My vision is that if the num. of spikes is less than a particular threshold (zero, one or two - you pick the threshold!), then we must return NaN, followed by a warning that a NaN is returned because there are not enough spikes. If this is the case, we don't need `with_nan` argument at all!

That is my concern. We'll discuss this on Tue morning. In the meanwhile, what do you think?

Reply:
Alright, we agreed on the following:
* remove with_nan
* if n_spikes < 2:
   warning.warn(blah-blah)
   return np.nan

because if the num. of spikes is less than 2, it does not make sense to compute CCH anyway.

Reply:
Hey @dizcza! Thanks for reviewing!

I will implement the changes right now. 

I only have one major concern, I would not raise a warning for this, this to me is expected behavior and I would not want my terminal flooded with warnings when I am applying this function to hundreds of slices of data.

Or does warnings have some fancy option to only warn for the same thing once in the same session?

Reply:
yes, use `warnings.simplefilter('once', UserWarning)` either inside the function or in the module (py file) itself - need to test. If inside the function works, leave it there - just before calling warnings.warn.

Reply:
I am quite sure it would not work from inside the function (due to multiple calls from different executions), but I will add it to my workflow and that will hopefully work!